core/api/src/graphql/public/root/query/ln-invoice-payment-status-by-hash.ts

```diff
@@ -0,0 +1,43 @@
+import { Lightning } from "@/app"
+import { WalletInvoiceStatus } from "@/domain/wallet-invoices"
+
+import { GT } from "@/graphql/index"
+import { mapAndParseErrorForGqlResponse } from "@/graphql/error-map"
+import LnInvoicePaymentStatusPayload from "@/graphql/public/types/payload/ln-invoice-payment-status"
+import LnInvoicePaymentStatusByHashInput from "@/graphql/public/types/object/ln-invoice-payment-status-by-hash-input"
+
+const LnInvoicePaymentStatusByHashQuery = GT.Field({
+  type: GT.NonNull(LnInvoicePaymentStatusPayload),
+  args: {
+    input: { type: GT.NonNull(LnInvoicePaymentStatusByHashInput) },
+  },
+  resolve: async (_, args) => {
+    const { paymentHash } = args.input
+    if (paymentHash instanceof Error) {
+      return { errors: [{ message: paymentHash.message }] }
```

Comment:
I believe we usually throw for queries and return only for mutations

Reply:
yes, I copied the structure used in the original status query.. please check https://github.com/GaloyMoney/galoy/pull/3899#discussion_r1484314966